### PR TITLE
Tca 55 modify the folder structure

### PR DESCRIPTION
**Pull request by pscacchi**
New folder structure:
-main
- presentation
  - home (activities | fragments | viewmodels)
  - login (activities | fragments | viewmodels)
- domain
- data
- di

The test folder were reordered too.

### CODE REVIEW

File: app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDataMapper.kt [hunk 1]
```diff
@@ -1,4 +1,7 @@
-package ar.scacchipa.twittercloneapp.data
+package ar.scacchipa.twittercloneapp.data.repository
+
+import ar.scacchipa.twittercloneapp.data.IMapper
+import ar.scacchipa.twittercloneapp.data.datasource.UserAccessTokenData
 
 class UserAccessTokenDataMapper: IMapper<UserAccessTokenData, UserAccessTokenDomain> {
```
Comment (Diana-Endava) [inline app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDataMapper.kt:6]: this file should be at the data level not Repository

File: app/src/main/java/ar/scacchipa/twittercloneapp/presentation/authwebdialog/AuthWebDialogViewModel.kt [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 import android.webkit.WebViewClient
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import ar.scacchipa.twittercloneapp.data.ResponseDomain
+import ar.scacchipa.twittercloneapp.data.repository.ResponseDomain
 import ar.scacchipa.twittercloneapp.domain.AuthorizationUseCase
 import ar.scacchipa.twittercloneapp.domain.ConsumableAuthUseCase
 import ar.scacchipa.twittercloneapp.utils.Constants
```
Comment (Diana-Endava) [inline app/src/main/java/ar/scacchipa/twittercloneapp/presentation/authwebdialog/AuthWebDialogViewModel.kt:12]: this comment apply for the whole folder "authwebdialog". This is part of the login so it should be inside the login folder and again the **naming** the name that has now, implies that is some kind of web dialog that can be reused anywhere in the app and it's not what it is at all, maybe try something like "loginWebSection" or "loginWebview" etc

File: app/src/main/java/ar/scacchipa/twittercloneapp/presentation/LoginCloneActivity.kt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package ar.scacchipa.twittercloneapp.component
+package ar.scacchipa.twittercloneapp.presentation
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
```
Comment (Diana-Endava) [inline app/src/main/java/ar/scacchipa/twittercloneapp/presentation/LoginCloneActivity.kt:4]: What is this?? if is a login related, should be inside the login folder, also the **name** why Clone? what are you cloning ???

File: app/src/main/java/ar/scacchipa/twittercloneapp/presentation/MainCloneActivity.kt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package ar.scacchipa.twittercloneapp.component
+package ar.scacchipa.twittercloneapp.presentation
 
 import android.os.Bundle
 import androidx.appcompat.app.ActionBar
```
Comment (Diana-Endava) [inline app/src/main/java/ar/scacchipa/twittercloneapp/presentation/MainCloneActivity.kt:4]: the **name again** why Clone? what are you cloning ???

File: app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDataMapper.kt (hunk 1 above)
Comment (Diana-Endava) [reply app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDataMapper.kt:6]: inside data, you are missing like a model folder where you'll put all these kinds of file that are exactly that, the models used by the data layer

File: app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDomain.kt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package ar.scacchipa.twittercloneapp.data
+package ar.scacchipa.twittercloneapp.data.repository
 
 data class UserAccessTokenDomain(
```
[the remark this answers is not in the document]
Comment (Diana-Endava) [reply app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDomain.kt:3]: like in the data layer, inside domain you are missing the models folder where you'll put all these kinds of files that are exactly that, the models used by the domain layer

**Comment by Diana-Endava** [comment]
Remember please to update the folder structure in the README

File: app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDataMapper.kt (hunk 1 above)
Comment (pscacchi) [reply app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDataMapper.kt]: A model folder inside the data folder was created and the POJO class was put in this.

File: app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDomain.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (pscacchi) [reply app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/UserAccessTokenDomain.kt]: Inside the domain folder, another model folder was created. The ResponseDomain class and UserAccessTokenDoman were moved to that folder.

File: app/src/main/java/ar/scacchipa/twittercloneapp/presentation/authwebdialog/AuthWebDialogViewModel.kt (hunk 2 above)
Comment (pscacchi) [reply app/src/main/java/ar/scacchipa/twittercloneapp/presentation/authwebdialog/AuthWebDialogViewModel.kt]: AuthWebDialog was converted to LoginWebSection

File: app/src/main/java/ar/scacchipa/twittercloneapp/presentation/LoginCloneActivity.kt (hunk 3 above)
Comment (pscacchi) [reply app/src/main/java/ar/scacchipa/twittercloneapp/presentation/LoginCloneActivity.kt]: TCA-55 The word 'Clone' was removed from the Activity identifiers

File: app/src/main/java/ar/scacchipa/twittercloneapp/presentation/MainCloneActivity.kt (hunk 4 above)
Comment (pscacchi) [reply app/src/main/java/ar/scacchipa/twittercloneapp/presentation/MainCloneActivity.kt]: The word 'Clone' was removed from the Activity identifiers

File: app/src/main/java/ar/scacchipa/twittercloneapp/presentation/LoginActivity.kt [hunk 6]
```diff
@@ -1,10 +1,10 @@
-package ar.scacchipa.twittercloneapp.component
+package ar.scacchipa.twittercloneapp.presentation
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 import ar.scacchipa.twittercloneapp.databinding.ActivityLoginLayoutBinding
 
-class LoginCloneActivity : AppCompatActivity() {
+class LoginActivity : AppCompatActivity() {
```
Comment (Diana-Endava) [inline app/src/main/java/ar/scacchipa/twittercloneapp/presentation/LoginActivity.kt:7]: move this to the corresponding login folder (the webview section or the other one), no reason to be outside

Comment (pscacchi) [reply app/src/main/java/ar/scacchipa/twittercloneapp/presentation/LoginActivity.kt]: A new folder level was added to the presentation folder structure.
- / presentation
  - / start
    - / splash
    - / login
  - / main
    - / home
    - / search

**Comment by sebastianMullerEndava** [comment]
> 
I dont like much that last added level of folders inside the presentation layer, I would not use main and starter folders.

File: README.md [hunk 7]
```diff
@@ -113,21 +113,36 @@ Asynchronous task are handled with [coroutines](https://developer.android.com/ko
...
-                        /datasource
+                    / data
+                        / model
+                        / sourcedata
+                        / repository
+                    / domain
+                        / model
+                        / usecase
+                    / presentation
+                       / start
```
Comment (sebastianMullerEndava) [inline README.md:124]: please update de readme.

Comment (pscacchi) [reply README.md]: The README was updated. Thes folder structure description was fixed.

**Review by sebastianMullerEndava — APPROVED** [review]

**Review by Diana-Endava — APPROVED** [review]

**Pull request merged**